Hublsoft.Net.LoginLogout.DataAccess/RegisteredUsersRepository.cs

```diff
@@ -105,5 +105,65 @@ public async Task IncrementFailedLoginAttemptsAsync(int id)
                 await con.CloseAsync();
             }
         }
+
+        public async Task<int> GetFailedLoginAttempsAsync(int id)
+        {
+            if (id == 0)
+            {
+                throw new Exception("Id cannot be 0");
+            }
+
+            var attempCount = 0;
+
+            try
+            {
+                using (var con = new MySqlConnection(ConnectionString))
+                {
+                    await con.OpenAsync();
+
+                    var stm = $"SELECT FailedLoginAttempts FROM RegisteredUsers WHERE Id = {id} ;";
```

Comment:
Best to wrap the sql statement with `SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED ;` and `SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ ;` so there are no record locks created around the record with the SELECT statement (similar to using `WITH (NOLOCK)` in SQL Server).

Reply:
I think in this instance it is probably okay to use read uncommitted to reduce the potential for blocking. We can accept the trade-off of reading a potentially uncommitted login attempt count